src/Suave/Suave.fsproj

```diff
@@ -140,15 +179,7 @@
     <Reference Include="System.Web.Extensions" />
     <Reference Include="System.Xml" />
   </ItemGroup>
-  <Choose>
-    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">
-      <ItemGroup>
-        <Reference Include="FSharp.Core">
-          <HintPath>..\..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
-          <Private>True</Private>
-          <Paket>True</Paket>
-        </Reference>
-      </ItemGroup>
-    </When>
-  </Choose>
+  <Target Name="BeforeBuild" BeforeTargets="Build" Condition="'$(TargetPlatform)' == 'coreclr'">
+    <Exec Command="..\..\tools\nuget\NuGet.exe restore -Source https://www.myget.org/F/dotnet-core;https://www.myget.org/F/fsharp-daily;https://nuget.org/api/v2 project.json -PackagesDirectory ..\..\packages -Verbosity detailed" />

```

Comment:
What’s worrying to you?

The new world will be a lot freer than the old world.  Although I hope you ensure that paket supports multiple versions of the same package in a single packages store, I.e. put back the version numbers.  I certainly will want to be able to build multiple tools dependent on different versions of specific contracts.  Not being able to do that would be problematic, requiring that I build and deploy everything as a monolithic lump – all tools referencing the same versions of contracts, of course I may choose to do that, but I want it to be an the option rather than a rule.

Also I don’t care about typing the long hint path, because an msbuild task takes care of that for me, and so I have to type nothing.

Etc …

BTW … the coreclr port of FSharp … works in a similar way to this, in-fact, that is where I copied (Ahem … reused) the deploy task from ☺

Kevin

From: Steffen Forkmann [mailto:notifications@github.com]
Sent: Thursday, November 12, 2015 11:00 AM
To: SuaveIO/suave <suave@noreply.github.com>
Cc: Kevin Ransom <Kevin.Ransom@microsoft.com>
Subject: Re: [suave] Enable build and run on coreclr for suave (#334)


In src/Suave/Suave.fsproj<https://github.com/SuaveIO/suave/pull/334#discussion_r44698853>:

> @@ -140,15 +179,7 @@

>      <Reference Include="System.Web.Extensions" />

>      <Reference Include="System.Xml" />

>    </ItemGroup>

> -  <Choose>

> -    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">

> -      <ItemGroup>

> -        <Reference Include="FSharp.Core">

> -          <HintPath>..\..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>

> -          <Private>True</Private>

> -          <Paket>True</Paket>

> -        </Reference>

> -      </ItemGroup>

> -    </When>

> -  </Choose>

> +  <Target Name="BeforeBuild" BeforeTargets="Build" Condition="'$(TargetPlatform)' == 'coreclr'">

> +    <Exec Command="..\..\tools\nuget\NuGet.exe restore -Source https://www.myget.org/F/dotnet-core;https://www.myget.org/F/fsharp-daily;https://nuget.org/api/v2<https://www.myget.org/F/dotnet-core;https:/www.myget.org/F/fsharp-daily;https:/nuget.org/api/v2> project.json -PackagesDirectory ..\..\packages -Verbosity detailed" />

Fast changes are not the issue for me here. It's more that things are starting to be labeled RC. And the version locking story is looking really really worrying. That's why I hope we can find a way out.

—
Reply to this email directly or view it on GitHub<https://github.com/SuaveIO/suave/pull/334/files#r44698853>.